### ISSUE TITLE
.noConflict

### ISSUE THREAD

**Issue by rmff** [js-cookie/js-cookie#128]
Hey guys, I'm trying to use noConflict but I get this message "caught TypeError: Cannot read property 'noConflict' of undefined", but when I use direct Cookies everything works. 

[Code](https://jsbin.com/wexena/4/edit?js,console)

**Comment by rmff** [comment]
var myObj = {

	transaction: {
		partial: false,
		partialEnd: false,
		trackerCookie: null
	},

	readToSend: false,
	readToSetTrackerCookie: false,
	readToRemoveTrackerCookie: false,

	/**
	 *
	 * @param window
	 */
	boot: function (window) {

		//Process queue
		this.processQueue(window[window['LLO']].queue);

		//Replace queue with LookLink.callMethod()
		window[window['LLO']] = this.callMethod;

		//Load Libs
		console.log("Load: jQuery");
		this.loadJQuery('https://ajax.googleapis.com/ajax/libs/jquery/1.12.0/jquery.min.js');
		console.log("Load: js Cookie");
		this.loadJSCookie('https://cdnjs.cloudflare.com/ajax/libs/js-cookie/2.1.0/js.cookie.min.js');

	}
};


/**
 * Return (this)
 *
 * @returns {myObj}
 */
myObj.getThis = function () {
	return this;
};		


/**
 * Load JavaScript Cookie v2.1.0 from CDN
 * https://github.com/js-cookie
 * Call setTrackerCookie on load
 */
myObj.loadJSCookie = function (url) {

	this.loadScript(url, true, function (LLO) {
		
		
		LLO.Cookies = window.Cookies.noConflict(); // NOT WORKING
		//LLO.Cookies = window.Cookies; // WORKS!!!
		
		
		if (LLO.readToSetTrackerCookie) {
			LLO.setTrackerCookie();
		}
		if (LLO.readToRemoveTrackerCookie) {
			LLO.removeTrackerCookie();
		}
	}(this.getThis()));
};

/**
 * Load jQuery from CDN or use loaded library
 * Call send() if transaction.readyToSend = true, on load
 */
myObj.loadJQuery = function (url) {

	if (typeof jQuery !== 'undefined') {
		this.jQuery = jQuery;

	} else {
		this.loadScript(url, true, function (LLO) {
			console.log("jQuery onload");

			LLO.jQuery = window.jQuery;

			if (LLO.jQuery !== 'undefined') {
				if (LLO.readToSend) {
					LLO.send();
				}
				return true;
			}else{
				return false;
			}
		}(this.getThis()));
	}
};

/**
 * Load script and call function on load
 *
 * @param url
 * @param scriptAsync
 * @param functionCallBackOnLoad
 */
myObj.loadScript = function (url, scriptAsync, functionCallBackOnLoad) {
	var script = document.createElement('script');
	var head = document.getElementsByTagName('head')[0], done = !1;
	script.async = scriptAsync;
	script.src = url;

	// Attach handlers for all browsers
	script.onload = script.onreadystatechange = function (functionCallBack){
		if (!done && (!document.readyState || document.readyState == 'loaded' || document.readyState == 'complete')) {
			done = functionCallBack();
			if(done) {
				script.onload = script.onreadystatechange = null;
			}
			//head.removeChild(script);
		}
	}(functionCallBackOnLoad);

	head.appendChild(script);
};

myObj.boot(window);

**Comment by FagnerMartinsBrack** [comment]
`.noConflict` is not going to work if the following conditions hold true:

```javascript
var isAMDEnvironment = typeof define === 'function' && define.amd;
var isCommonJSEnvironment = typeof exports === 'object';
```

Otherwise it should work.

Could you please create an [SSCCE](http://sscce.org/), removing all unnecessary code from your snippet and uploading the code to jsfiddle, jsbin or similar? Thanks.

**Comment by rmff** [comment]
Fagner,

I've uploaded the code, I'm sorry for post something big like that.

Both conditions are false in my case.
Did you know if there is some issue working using anonymous functions or use strict?

I've found some strange behavior, when I try to use noConflict inside javascript I get the error message, but if I do it using console, works.

As could see on [jsBin](https://jsbin.com/wexena/5/edit?js,console)

**Comment by FagnerMartinsBrack** [comment]
@rmff This is still not a Short, Self Contained example. There is a lot of unnecessary code there. Please read this: http://sscce.org

The lib works when loading through the network, here is an evidence: https://jsbin.com/seqanirice/1/edit?js,console

My only guess is that the code you provided might not be evaluating the loaded script, other than that I have no idea.

I would advise you to post on [StackOverflow under the tag `js-cookie`](http://stackoverflow.com/questions/tagged/js-cookie). It is much more likely to receive a correct response on why your code is not working there than posting here. Also, on stackoverflow there is better CEO and therefore more people can benefit from your question and responses.

**Comment by rmff** [comment]
Fagner,

I'm sorry if it is not SSCCE, but I needed to post some "unnecessary" code because the problem was inside that code, and not inside js-cookie lib.

I've found the solution, it's nothing related to js-cookie. It's related on how to pass inline anonymous functions, to other function.

I've tried to use the example bellow, and it's incorrect way to inject inline functions. Because that js-cookie lib didn't load.

```javascript
foo(args, function(myArg) {
	//some code...
}(someArg));
```
In this case foo is receiving the result of anonymous function and not the function itself.

The correct way is to be sure that someArg is in the scope and use someArg direct inside the anonymous function, like this:

```javascript
var someArg = 'some value';
foo(args, function () {
	//direct use of someArg
	console.log(someArg);	
});
```

Sorry bother you with a non js-cookie related problem
